flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCUpsertTableSinkITCase.java

```diff
@@ -252,4 +282,100 @@ public void testBatchSink() throws Exception {
 				Row.of("Bob", 1)
 		}, DB_URL, OUTPUT_TABLE3, new String[]{"NAME", "SCORE"});
 	}
+
+	@Test
+	public void testCreateTableIfNotExists() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		env.getConfig().enableObjectReuse();
+		env.getConfig().setParallelism(1);
+		StreamTableEnvironment tEnv = StreamTableEnvironment.create(env);
+
+		Table t = tEnv.fromDataStream(getFullTypeDataStream(env), "c0, c1, c2, c3, c4, c5, c6, c7, c8, c9");
+
+		tEnv.registerTable("T", t);
+
+		tEnv.sqlUpdate(
+			"CREATE TABLE upsertSink (" +
+				"  c0 BOOLEAN," +
+				"  c1 INTEGER," +
+				"  c2 BIGINT," +
+				"  c3 FLOAT," +
+				"  c4 DOUBLE," +
+				"  c5 DECIMAL," +
+				"  c6 STRING," +
+				"  c7 DATE," +
+				"  c8 TIME(0)," +
+				"  c9 TIMESTAMP(3)" +
+				") WITH (" +
+				"  'connector.type'='jdbc'," +
+				"  'connector.url'='" + DB_URL + "'," +
+				"  'connector.table'='" + OUTPUT_TABLE4 + "'," +
+				"  'connector.write.auto-create-table'='true'" +
+				")");
+
+		tEnv.sqlUpdate("INSERT INTO upsertSink " +
+			" SELECT c0, c1, c2, c3, c4, max(c5), max(c6), max(c7), max(c8), max(c9) " +
+			" FROM T WHERE c1 > 2" +
+			" group by c0, c1, c2, c3, c4");
+		tEnv.execute("testCreateTableIfNotExists");
+		check(new Row[] {
+				//JDBC treats SQL FLOAT/DOUBLE as java.lang.Double, convert float value to double before compare.
+				Row.of(false, 3, 1002L, Double.valueOf(0.1f), 12.26d, new BigDecimal("5.330000000000000000"), "To be or not to be 3",
+					Date.valueOf("2020-03-10"), Time.valueOf("00:00:03"), Timestamp.valueOf("2020-03-10 00:00:00.021")),
+				Row.of(false, 4, 1003L, Double.valueOf(0.11f), 12.27d, new BigDecimal("5.440000000000000000"), "To be or not to be 4",
+					Date.valueOf("2020-03-10"), Time.valueOf("00:00:04"), Timestamp.valueOf("2020-03-10 00:00:00.021"))},
+			DB_URL, OUTPUT_TABLE4, new String[]{"c0", "c1", "c2", "c3", "c4", "c5", "c6", "c7", "c8", "c9"});
+
+		clearAutoCreatedTable();
+	}
+
+	@Test
+	public void testCreateTableIfNotExistsUseBlinkPlanner() throws Exception {
```

Comment:
Please use `Parameterized` to test against different planners to avoid duplicate code, or just test blink planner.